File: apps/server/app/routes/simple_api.py
from fastapi import FastAPI,Depends,APIRouter
from sqlalchemy.orm import Session
from database import get_db,engine
from models import Nodes,UploadFileRecord, ImportedData,Link,family,world,Structure,Population


app=APIRouter()
@app.get('/data')
def get_datas(db:Session=Depends(get_db)):
    data = db.query(ImportedData).all()
    return [item.to_dict() for item in data]

@app.get('/nodes')
def get_nodes(db:Session=Depends(get_db)):
    nodes = db.query(Nodes).all()
    return nodes

@app.get('/population_structure')
def get_population_structure(db:Session=Depends(get_db)):
    data = db.query(Population).all()
    return data

@app.get('/base_nodes')
def get_base_nodes(db:Session=Depends(get_db)):
    data = db.query(Nodes).filter(Nodes.type=='基础需求').all()
    return data

@app.get('/maslow_needs')
def get_maslow_needs(db:Session=Depends(get_db)):
    data = db.query(Nodes).filter(Nodes.type=='马斯洛需求').all()
    return data

@app.get('/old_structure')
def get_old_structure(db:Session=Depends(get_db)):
    data = db.query(Structure).filter(Structure.type=='old').all()
    return data

@app.get('/new_structure')
def get_new_structure(db:Session=Depends(get_db)):
    data = db.query(Structure).filter(Structure.type=='new').all()
    return data

@app.get('/link')
def get_link(db:Session=Depends(get_db)):
    link = db.query(Link).all()
    return link

@app.get('/solar')
def get_solar(db:Session=Depends(get_db)):
    solar = db.query(solar).all()
    return solar

@app.get('/family')
def get_family(db:Session=Depends(get_db)):
    data = db.query(family).all()
    return data

@app.get('/world')
def get_world(db:Session=Depends(get_db)):
    data = db.query(world).all()
    return data

File: apps/web/src/api/simple_api.js

import request from '../util/request'
import axios from 'axios'

const url1='/routes/simple_api/nodes'
export function getNodes(){
    // return request.get('http://localhost:8000/books')
    return request({
        url:url1,
        method:'get'
    })    
}

const url_liveCycle='/routes/simple_api/liveCycle'
export function getLiveCycle(){
    // return request.get('http://localhost:8000/books')
    return request({
        url:url_liveCycle,
        method:'get'
    })    
}

const url2='/routes/simple_api/link'
export function getLink(){
    // return request.get('http://localhost:8000/books')
    return request({
        url:url2,
        method:'get'
    })    
}

const url3='/routes/simple_api/world'
export function getWorld(){
    // return request.get('http://localhost:8000/books')
    return request({
        url:url3,
        method:'get'
    })    
}


const url6='/routes/simple_api/family'
export function getFamily(){
    return request({
        url:url6,
        method:'get'
    })    
}


const url8='/routes/simple_api/base_nodes'
export function get_base_nodes(){
    return request({
        url:url8,
        method:'get'
    })    
}

const url9='/routes/simple_api/maslow_needs'
export function get_maslow_needs(){
    return request({
        url:url9,
        method:'get'
    })    
}

const old_structure_url='/routes/simple_api/old_structure'
export function get_old_structure(){
    return request({
        url:old_structure_url,
        method:'get'
    })    
}

const new_structure_url='/routes/simple_api/new_structure'
export function get_new_structure(){
    return request({
        url:new_structure_url,
        method:'get'
    })    
}


const solar='/routes/simple_api/solar'
export function get_solar(){
    return request({
        url:solar,
        method:'get'
    })    
}


const url_population_structure='/routes/simple_api/population_structure'
export function population_structure(year){
    return request({
        url:url_population_structure,
        method:'get'
    })    
}